### ISSUE TITLE
ECS `process` add TTY and ENV fields

### ISSUE THREAD

**Issue by BenB196** [elastic/ecs#1535]
**Summary**

I'd like to see the addition of fields for TTY (Linux TTY) and ENV (Linux environment variables) values within the process fields.

**Motivation**:

While these fields don't often come up, they do provide value, and there doesn't appear to be a clear place within ECS for these fields.

Capturing these additional pieces of information allow for more information regarding the event when provided.

**Detailed Design**:

Provide additional details around the design of the proposed changes.

* Field names:
  - `process.tty`
  - `process.env`
* Example values for the fields:
 - `process.tty`
   - `/dev/pts/0`
   - `/pts/93`
   - `unkown`
 - `process.env`
   - `JDK_HOME=/usr/lib64/jvm/java`
   - `USER=root`
* Suggested appropriate datatypes:
- `process.tty`
  - This field is technically a file path, so would probably just use the file path mappings:
  - `keyword`
  - Multi-field
    - `text`
- `process.env`
  - I'm not really sure the best way to map this field.
  - I could see it either being mapped as a `keyword` with a `text` multifield for simplicity
  - Or mapped as a `nested` object field as environment variables are technically Key Value Pairs.
* Any example events that map to the proposed use case(s):

Example event where these fields are useful.

`<85>Jul 22 13:03:30 $HOSTNAME sudo: root : TTY=unknown ; PWD=$PATH_TO_JAVA_SDK ; USER=root ; ENV=PYTHONPATH=$PATH_TO_BIN ; COMMAND=$COMMAND`

`$CAPS_WITH_UNDER` are placeholder text for real data that was snipped.

**Comment by ebeahan** [comment]
For `process.tty`, would a description like `the terminal from which the analyzed process was invoked` capture the field's intent?

I don't think we'd have to limit the scope to just Linux. MacOS and, I think, even recently [Windows](https://devblogs.microsoft.com/commandline/windows-command-line-introducing-the-windows-pseudo-console-conpty/) now have an implementation. Same with gathering the process environment data.

The `flattened` type might make sense for a `process.environment` field.

**Comment by ebeahan** [comment]
@nfritts @dcode I came across some past endpoint-focused discussions on collecting process environment variables. Thought I'd bring this issue to your attention and see if there's any overlap.

**Comment by nfritts** [comment]
We're working some of this for Mac specifically as a part of this issue: https://github.com/elastic/endpoint-dev/issues/6872


I think this comment is highly relevant: https://github.com/elastic/endpoint-dev/issues/6872#issuecomment-832215981


We have gone down that path for now.  Overall we plan to hard code exactly which environment variables we think are most relevant and have them stared as keywords like:
```
{
    "process": {
        "environment": {
            "DYLD_LIBRARY_PATH": "/tmp",
            "DYLD_INSERT_LIBRARIES": "inject.dylib"
        }
    }
}
```

There is a long sample in that issue, but it ends up looking like the snippet below.  without an update to ECS we will likely use the same format, but store it under the Ext block.

```
    "process": {
        "Ext": {
            "ancestry": [
                "ODUzNDYzMzMtMTAyNy1jMmI0LTQ3NmQtZmFkYzM3NTZkMDlmLTEzMjg4LTEzMjY3NjI1Mjg3Ljk0NTk4NzAwMA==" ]
        },
        "args": [
            "/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python",
            "-c",
            "print('random_string_test')"
        ],
        "args_count": 3,
        "command_line": "/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -c print('random_string_test')",
        "entity_id": "ODUzNDYzMzMtMTAyNy1jMmI0LTQ3NmQtZmFkYzM3NTZkMDlmLTEzMjg4LTEzMjY3NjI1Mjg3Ljk4MDgwNzAwMA==",
        "environment": {
            "dyld_insert_libraries": "Psi2juj.dylib",
            "dyld_library_path": "/tmp/",
            "tmpdir": "tmp.AWA4dTERha"
        },
```

**Comment by ebeahan** [comment]
Both `process.tty.*` and `process.env_vars` fields will be included in 8.2:  https://www.elastic.co/guide/en/ecs/8.2/ecs-process.html#field-process-env-vars.

These fields were added as part of the [Linux Event Model RFC](https://github.com/elastic/ecs/blob/main/rfcs/text/0030-linux-event-model.md), and examples populating these fields are captured there.
